code/modules/admin/DB ban/functions.dm

```diff
@@ -95,10 +59,9 @@
 		a_computerid = src.owner:computer_id
 		a_ip = src.owner:address
 
-	if(blockselfban)
-		if(a_ckey == ckey)
-			usr << "<span class='danger'>You cannot apply this ban type on yourself.</span>"
-			return
+	if(!job && (a_ckey == ckey))
+		usr << "<span class='danger'>You cannot server ban yourself.</span>"

```

Comment:
Actually, I'd recommend removing that. I've had admins (Niknak) try to temp admin ban themselves to study and shit, and have to bug another admin to do it.